Codigo/Declares.bas

```diff
@@ -2690,3 +2641,77 @@ Public Pasos()               As tPaso
 Public DBError As String
 
 Public EnEventoFaccionario As Boolean
+
+
+Public Enum e_EffectOverTimeType
+    eModifyHelath = 1
+    [EffectTypeCount]
+End Enum
+
+Public Enum e_EOTTargetLimit
+    eSingle = 1 'Only one on target for this type
+    eSingleByCaser 'The target can have more than 1 effect of this type but only 1 for every caster
+    eAny 'No limits
+End Enum
+
+Public Type t_BaseDotInfo
+    TargetRef As t_AnyReference
+    UniqueId As Integer
+    RemoveEffect As Boolean
+    EotType As e_EffectOverTimeType
+    EotId As Integer
+End Type
+
+
+Public Function IsValidRef(ByRef Ref As t_AnyReference) As Boolean
+    IsValidRef = False
+    If Ref.ArrayIndex < LBound(UserList) Then
```

Comment:
well the idea is to simplify the test, otherwise we would need to write a lot of code for both cases as we do now everywhere with TargetNPC and TargetUser

We already have isValidUserRef and isvaludNpcrRef for reference that can only hold one type